[WIP][mono] Basic SIMD support for System.Numerics.Vector2 on arm64

matouskozak opened the pull request:
This PR adds basic SIMD support for `System.Numerics.Vector2` on arm64. Equaling the current support present for `System.Numerics.Vector4`.

Current SIMD support for `Vector2` with mini/llvm:
- SN_ctor
- SN_Abs
- SN_Add
- SN_Clamp
- SN_Divide (currently disabled `Vector2` / `float` scenario, will enable in the next PR)
- SN_Dot
- SN_Max
- SN_Min
- SN_Multiply
- SN_Negate
- SN_SquareRoot
- SN_Subtract
- SN_get_Item
- SN_get_One
- SN_get_UnitX
- SN_get_UnitY
- SN_get_Zero
- SN_op_Addition
- SN_op_Division
- SN_op_Equality
- SN_op_Inequality
- SN_op_Multiply
- SN_op_Subtraction
- SN_op_UnaryNegation
- SN_set_Item

Future work on the missing intrinsic is tracked here https://github.com/dotnet/runtime/issues/91394.
Contributes to: https://github.com/dotnet/runtime/issues/73462

---
p.s. These getters currently use 128-bit code paths for emitting const values (`emit_xconst_v128`) even for Vector2 (64-bit vector):
- SN_get_UnitX
- SN_get_UnitY
- SN_get_One

While it looks like the generated code is okay locally, it could potentially cause problems in `mono_arch_emit_exceptions` or in subsequent exception handling https://github.com/dotnet/runtime/blob/c12429abaa1762dc971d6e4ef2f34e7140c1b364/src/mono/mono/mini/mini-arm64.c#L6092-L6094

@fanyang-mono @jandupej Any thoughts on that? Is dedicated 64-bit const value emit worth implementing?

matouskozak committed:
enable Vector2 on arm64

matouskozak committed:
enable Vector2 on arm64

matouskozak committed:
Vector2 fabs arm64 instrinsics

matouskozak committed:
Vector2 fabs arm64 instrinsics

matouskozak committed:
Vector2 fmin and fmax arm64 intrinsics

matouskozak committed:
Vector2 fmin and fmax arm64 intrinsics

matouskozak committed:
Vector 2 fdiv arm64 intrinsics

matouskozak committed:
Vector 2 fdiv arm64 intrinsics

matouskozak committed:
Vector2 Dot product intrinsics arm64

matouskozak committed:
Vector2 Dot product intrinsics arm64

matouskozak committed:
Vector2 sub and sqrt intrinsics support for arm64

matouskozak committed:
Vector2 sub and sqrt intrinsics support for arm64

matouskozak committed:
Vector2 op_equality fix or arm64

matouskozak committed:
Vector2 op_equality fix or arm64

matouskozak committed:
Support

matouskozak committed:
Support

matouskozak commented:
More paths where 128-bit vector paths are used detected (e.g. `OP_XZERO`). Changing to WIP and I'll investigate further.

matouskozak committed:
OP_XZERO 64-bit vector support arm64

matouskozak committed:
OP_XZERO 64-bit vector support arm64

jandupej commented:
> These getters currently use 128-bit code paths for emitting const values...

You can use a `fmov` to flood the lower two floats with `1.0f`. This gives you the fastest `SN_get_One` possible (there is a 64-bit variant of this, with `q=0`). To make `SN_get_UnitX/Y` you can shift the vector left or right as doubles by 32. Zeros are shifted in, so this will give you a `(0.0f, 1.0f)` or reverse. This will destroy the upper 64 bits of the register, but it shouldn't be a problem as only the lower 64 bits are of importance.

ViktorHofer committed:
Fix official builds for NativeAOT (#91454)

The ILLink ProjectReference output shouldn't be copied to consuming projects.

From the docs:
> Private	Optional boolean. Specifies whether the reference should be copied to the output folder. This attribute matches the Copy Local property of the reference that's in the Visual Studio IDE.

Commit:
[ARM64] Add g_GCShadowEnd to JIT_WriteBarrier_Table (#91342)

* [ARM64] Add g_GCShadowEnd to JIT_WriteBarrier_Table

This change moves address of g_GCShadowEnd to JIT_WriteBarrier_Table like
others variables used in Write Barrier.

This fix simmilar to RISC-V one https://github.com/dotnet/runtime/pull/90036

* [ARM64] Move GCShadow vars to the end of the wbs block

* Update src/coreclr/vm/arm64/asmhelpers.asm

---------

Co-authored-by: Jan Kotas <jkotas@microsoft.com>

campersau committed:
[browser] [wasm] Request Streaming upload (#91295)

Co-authored-by: pavelsavara <pavel.savara@gmail.com>

huoyaoyuan committed:
Remove some unused headers and functions (#91177)

* Remove winternl.h

* Remove winver.h

* Remove verrsrc.h

* Remove dbghelp.h

* Remove conio.h

* Remove io.h

* FILECFAbsoluteTimeToFileTime

* Delete some unused header files

* More unused structs and macros

* Update fxver.h

pavelsavara committed:
[browser] make sure that no errors are lost (#91320)

matouskozak committed:
OP_EXPAND/EXTRACT_R4/R8 for 64 bit vector arm64

matouskozak committed:
OP_EXPAND/EXTRACT_R4/R8 for 64 bit vector arm64

matouskozak committed:
Revert "OP_EXPAND/EXTRACT_R4/R8 for 64 bit vector arm64"

This reverts commit 27d244b2fcc28bfd7314deab2d31b192c8d3df76.

matouskozak committed:
Revert "OP_EXPAND/EXTRACT_R4/R8 for 64 bit vector arm64"

This reverts commit 27d244b2fcc28bfd7314deab2d31b192c8d3df76.

akoeplinger committed:
Improve DevContainer configuration for Codespaces (#91465)

Adds a default configuration that comes with all the dependencies preinstalled, but not prebuilt. This reduces confusion if you don't use the "New with options..." dropdown when creating the Codespace since you'd just get a standard Ubuntu container then.

Also install a specific version of dotnet-serve.

hez2010 committed:
JIT: Remove the minopts policy against cctors (#90792)

* Introduce a switch to use FullOpts for cctors

* Remove the obsolete policy around cctors

* Revert changes around the new switch

hamarb123 committed:
Fix `Item4` is missing in some `ValueTuple`s' `IStructuralEquatable.Equals` (#91461)

* Add missing comparisons

* Add unit tests

jkoritzinsky committed:
Disable analyzers to fix the workloads build. (#91476)

radical committed:
[wasm] Fix perf pipeline (#91446)

* workloads: Skip updating the targeting pack

This is required because the runtime pack is still 8.0, but updating the
targeting pack from the local build would make it 9.0, thus causing this
error for non-wasm apps:

```
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.

File name: 'System.Runtime, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
```

* [wasm] makefile cleanup

* [wasm] Enable non-wasm build tests

elinor-fung committed:
Add comment about support for null context in `hostfxr_get_runtime_delegate` (#91479)

carlossanlop committed:
Bump Microsoft.Private.Intellisense package version (#91285)

sbomer committed:
Share more code between ILLink/ILCompiler test infra (#91358)

Uses the code sharing infra added in
24e5bce991950c514ba6d062324ced2a7d62c634. This shares most of the
logic that was easily shareable in the `TestCasesRunner`
directory.

cshung committed:
Opportunistically use stack allocation for GC dynamic events (#91001)

lambdageek committed:
Bump mono to -std=gnu11 (#91421)

Change the mono build to require C11 (with gnu extensions on gcc/clang platforms).
* Change `g_static_assert` to be `_Static_assert` or `static_assert` as apropriate.
* Change `_DN_STATIC_ASSERT` to be `static_assert`
* Add static asserts in `jiterp.c` when it casts between `T*` and `atomic_T*`
* Add C11 guidance to the mono coding guide doc

Contributes to #90404 

---

* Bump mono to -std=gnu99; use static_assert

* don't fall back to runtime checks for g_static_assert

* fix static assert that wasn't a constant expression

* use static_assert in shared containers

* bump C standard in offsets-tool.py

* use _Static_assert before C23

   Dont' include assert.h in glib.h because some of our older 3P code includes assert.h on its own and there are conflicts

* use CMAKE_C_STANDARD and related properties

* jiterp: static_assert that atomic ops are (less likely) to go wrong

   Not every C implementation guarantees that atomic operations on arbitrary types are lock free.  So for example, casting between atomic_ushort* and uint16_t* might not actually be ok to do.  We can't assert that they're inter-castable, but at least assert that they're the same size and that atomic_ushort is always lock-free. There might still be restrictions (for example atomic_ushort might have to be aligned differently) but this should at least catch obvious data corruption.

* Add C11 guidance to the Mono coding guide

* jiterp: long is 32-bits on wasm; use llong

Youssef1313 committed:
Make `--skip-unresolved` the default when not running from SDK (#90512)

* Make `--skip-unresolved` the default when not running from SDK

* Fix command-line doc

* Update NullableOnConstraints test

* chore: Update CanPreserveNullableCustomOperators test

* Remove `--skip-unresolved` from targets as it's the default now in command-line linker

* Fix tests that check --skip-unresolved false behavior

---------

Co-authored-by: Sven Boemer <sbomer@gmail.com>

jtschuster committed:
Check for generic structs in "needs marshalling" check (#90836)

Co-authored-by: Aaron Robinson <arobins@microsoft.com>
Co-authored-by: Jeremy Koritzinsky <jkoritzinsky@gmail.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/arcade dotnet/xharness dotnet/emsdk dotnet/sdk (#91323)

* Update dependencies from https://github.com/dotnet/arcade build 20230829.1

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 8.0.0-beta.23428.2 -> To Version 8.0.0-beta.23429.1

Dependency coherency updates

Microsoft.DotNet.XliffTasks
 From Version 1.0.0-beta.23423.1 -> To Version 1.0.0-beta.23426.1 (parent: Microsoft.DotNet.Helix.Sdk

* Update dependencies from https://github.com/dotnet/xharness build 20230829.1

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Common , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 8.0.0-prerelease.23421.1 -> To Version 8.0.0-prerelease.23429.1

* Update dependencies from https://github.com/dotnet/emsdk build 20230829.3

Microsoft.NET.Workload.Emscripten.Current.Manifest-9.0.100.Transport
 From Version 9.0.0-alpha.1.23427.1 -> To Version 9.0.0-alpha.1.23429.3

* Update dependencies from https://github.com/dotnet/sdk build 20230830.6

Microsoft.DotNet.ApiCompat.Task
 From Version 9.0.100-alpha.1.23429.3 -> To Version 9.0.100-alpha.1.23430.6

* Dependency coherency updates

runtime.linux-arm64.Microsoft.NETCore.Runtime.ObjWriter,runtime.linux-x64.Microsoft.NETCore.Runtime.ObjWriter,runtime.linux-musl-arm64.Microsoft.NETCore.Runtime.ObjWriter,runtime.linux-musl-x64.Microsoft.NETCore.Runtime.ObjWriter,runtime.win-arm64.Microsoft.NETCore.Runtime.ObjWriter,runtime.win-x64.Microsoft.NETCore.Runtime.ObjWriter,runtime.osx-arm64.Microsoft.NETCore.Runtime.ObjWriter,runtime.osx-x64.Microsoft.NETCore.Runtime.ObjWriter,runtime.linux-arm64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.linux-x64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.linux-musl-arm64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.linux-musl-x64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.win-arm64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.win-x64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.osx-arm64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.osx-x64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.linux-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.linux-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.linux-musl-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.linux-musl-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.linux-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.linux-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.linux-musl-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.linux-musl-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.win-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.win-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.osx-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.osx-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.osx-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.osx-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools
 From Version 16.0.5-alpha.1.23414.1 -> To Version 16.0.5-alpha.1.23423.2 (parent: Microsoft.NET.Workload.Emscripten.Current.Manifest-9.0.100.Transport

* Update dependencies from https://github.com/dotnet/sdk build 20230831.2

Microsoft.DotNet.ApiCompat.Task
 From Version 9.0.100-alpha.1.23429.3 -> To Version 9.0.100-alpha.1.23431.2

* Update dependencies from https://github.com/dotnet/arcade build 20230831.2

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 8.0.0-beta.23428.2 -> To Version 8.0.0-beta.23431.2

Dependency coherency updates

Microsoft.DotNet.XliffTasks
 From Version 1.0.0-beta.23423.1 -> To Version 1.0.0-beta.23426.1 (parent: Microsoft.DotNet.Helix.Sdk

* Update dependencies from https://github.com/dotnet/xharness build 20230831.1

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Common , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 8.0.0-prerelease.23421.1 -> To Version 8.0.0-prerelease.23431.1

* Update dependencies from https://github.com/dotnet/sdk build 20230831.39

Microsoft.DotNet.ApiCompat.Task
 From Version 9.0.100-alpha.1.23429.3 -> To Version 9.0.100-alpha.1.23431.39

* Dependency coherency updates

runtime.linux-arm64.Microsoft.NETCore.Runtime.ObjWriter,runtime.linux-x64.Microsoft.NETCore.Runtime.ObjWriter,runtime.linux-musl-arm64.Microsoft.NETCore.Runtime.ObjWriter,runtime.linux-musl-x64.Microsoft.NETCore.Runtime.ObjWriter,runtime.win-arm64.Microsoft.NETCore.Runtime.ObjWriter,runtime.win-x64.Microsoft.NETCore.Runtime.ObjWriter,runtime.osx-arm64.Microsoft.NETCore.Runtime.ObjWriter,runtime.osx-x64.Microsoft.NETCore.Runtime.ObjWriter,runtime.linux-arm64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.linux-x64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.linux-musl-arm64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.linux-musl-x64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.win-arm64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.win-x64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.osx-arm64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.osx-x64.Microsoft.NETCore.Runtime.JIT.Tools,runtime.linux-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.linux-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.linux-musl-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.linux-musl-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.linux-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.linux-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.linux-musl-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.linux-musl-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.win-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.win-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.osx-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.osx-arm64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools,runtime.osx-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Sdk,runtime.osx-x64.Microsoft.NETCore.Runtime.Mono.LLVM.Tools
 From Version 16.0.5-alpha.1.23414.1 -> To Version 16.0.5-alpha.1.23423.2 (parent: Microsoft.NET.Workload.Emscripten.Current.Manifest-9.0.100.Transport

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

AaronRobinsonMSFT committed:
Fix usage of load library (#91483)

* Use LoadLibraryExW on Windows.

dotnet-maestro[bot] committed:
[main] Update dependencies from dnceng/internal/dotnet-optimization (#91203)

* Update dependencies from https://dev.azure.com/dnceng/internal/_git/dotnet-optimization build 20230827.2

optimization.linux-arm64.MIBC.Runtime , optimization.linux-x64.MIBC.Runtime , optimization.windows_nt-arm64.MIBC.Runtime , optimization.windows_nt-x64.MIBC.Runtime , optimization.windows_nt-x86.MIBC.Runtime , optimization.PGO.CoreCLR
 From Version 1.0.0-prerelease.23426.6 -> To Version 1.0.0-prerelease.23427.2

* Update dependencies from https://dev.azure.com/dnceng/internal/_git/dotnet-optimization build 20230829.4

optimization.linux-arm64.MIBC.Runtime , optimization.linux-x64.MIBC.Runtime , optimization.windows_nt-arm64.MIBC.Runtime , optimization.windows_nt-x64.MIBC.Runtime , optimization.windows_nt-x86.MIBC.Runtime , optimization.PGO.CoreCLR
 From Version 1.0.0-prerelease.23426.6 -> To Version 1.0.0-prerelease.23429.4

* Update dependencies from https://dev.azure.com/dnceng/internal/_git/dotnet-optimization build 20230830.6

optimization.linux-arm64.MIBC.Runtime , optimization.linux-x64.MIBC.Runtime , optimization.windows_nt-arm64.MIBC.Runtime , optimization.windows_nt-x64.MIBC.Runtime , optimization.windows_nt-x86.MIBC.Runtime , optimization.PGO.CoreCLR
 From Version 1.0.0-prerelease.23426.6 -> To Version 1.0.0-prerelease.23430.6

* Update dependencies from https://dev.azure.com/dnceng/internal/_git/dotnet-optimization build 20230831.4

optimization.linux-arm64.MIBC.Runtime , optimization.linux-x64.MIBC.Runtime , optimization.windows_nt-arm64.MIBC.Runtime , optimization.windows_nt-x64.MIBC.Runtime , optimization.windows_nt-x86.MIBC.Runtime , optimization.PGO.CoreCLR
 From Version 1.0.0-prerelease.23426.6 -> To Version 1.0.0-prerelease.23431.4

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/source-build-reference-packages build 20230901.1 (#91493)

Microsoft.SourceBuild.Intermediate.source-build-reference-packages
 From Version 8.0.0-alpha.1.23424.1 -> To Version 8.0.0-alpha.1.23451.1

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

Maoni0 committed:
Cleanup only changes (#91445)

Cleanup only changes

+ fixed compiler errors for when BACKGROUND_GC is not defined

+ made a few things only defind when USE_REGIONS is not defined as they are only used for segments case

+ fixed an AV in a dprintf in descr_generations that @cshung noticed

+ reverted the dprintf change I accidently left in a previous PR that @mrsharm noticed

+ got rid of some unused methods

+ got rid of some useless, obsolete or wrong comments and move the finalize lock comment to an appropriate place

+ fixed some inappropriate line width, inconsistent formatting, unused args/fields and typos
  there are other inconsistent formating like spacing but there are many so they should be in their own PR to not completely dwarf changes in this one.

EgorBo committed:
Fix impNonConstFallback (#91345)

kunalspathak committed:
Fix the VN for xor operation (#91503)

github-actions[bot] committed:
Automated bump of chrome version (#91512)

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/installer build 20230901.3 (#91502)

clamp03 committed:
[RISC-V] Implement ThisPtrRetBufPrecode::Init (#91451)

SkiFoD committed:
#78303 Add transformation ~v1 & v2 to VectorXxx.AndNot(v1, v2) (#81993)

Co-authored-by: EgorBo <egorbo@gmail.com>

campersau committed:
[browser][HTTP] remove special StringContent handling (#91463)

YohDeadfall committed:
Unsigned check of index inside ElementAt (#91522)

layomia committed:
Enable interceptors implicitly in binder gen nupkg when generator is enabled (#91469)

* Enable interceptors implicitly in binder gen nupkg when generator is enabled

* Simplify impl & remove unneeded target

liveans committed:
Activate Quic Alpn Narrowing Test (#91116)

* Delete ActiveIssue

* Disable Not Existing Inline Data

* Delete connection accept

* Merge assert and variable

* Add listener throws case

* Update src/libraries/System.Net.Quic/tests/FunctionalTests/QuicListenerTests.cs

Co-authored-by: Marie Píchová <11718369+ManickaP@users.noreply.github.com>

---------

Co-authored-by: Marie Píchová <11718369+ManickaP@users.noreply.github.com>

Thefrank committed:
Add FreeBSD case to gcenv. (#91524)

Closes #64935

jkotas committed:
Delete unused methods from CoreCLR PAL (#91554)

janvorli committed:
Fix NativeAOT unhandled exception stack trace (#91415)

* Fix NativeAOT unhandled exception stack trace

The recent change to add a new exception handling mechanism to coreclr
has broken stack traces on unhandled exceptions in nativeaot.

This change fixes it by reverting the part of the change that caused the
problem and that turned out to not to be needed for the new exception
handling either.

Close #91298

* Add unhandled exception test

* Reflect PR feedback

* Fix Mono console log parsing

elinor-fung committed:
Remove unused CLR_E_BIND_* errors (#91488)

LuckyXu-HF committed:
[LoongArch64, RISC-V] Fix the assertion error for MethodTable::GetLoongArch64PassStructInRegisterFlags() and MethodTable::GetRiscv64PassStructInRegisterFlags(). (#91570)

LoopedBard3 committed:
[PERF] Use the major version for the net_.0 version in create mono dotnet step (#91452)

* Use the major version from the Versions.props for the net_.0 version in create mono dotnet.

* Setup testing.

* Revert "Setup testing."

This reverts commit c948467b92254e119536fc5e1a43c57841790108.

* Try updating and using framework instead.

* Update add_properties_to_pipeline.proj

pavelsavara committed:
fix https://github.com/dotnet/runtime/issues/91584 (#91585)

matouskozak committed:
fix get/set_Item for Vector2

matouskozak committed:
fix get/set_Item for Vector2

tannergooding committed:
Adding a naive implementation of various primitive tensor operations (#91228)

* Adding a naive implementation of various primitive tensor operations

* Adding tests covering the new tensor primitives APIs

* Adding tensor primitives APIs to the ref assembly

* Allow .NET Framework to build/run

* Sync TFMs between ref and src, csproj simplication and clean-up

* Apply suggestions from code review

Co-authored-by: Viktor Hofer <viktor.hofer@microsoft.com>

* Don't use var

* Fix the S.N.Tensors readme and remove the file marking it as non-shipping

---------

Co-authored-by: Viktor Hofer <viktor.hofer@microsoft.com>
Co-authored-by: Michael Sharp <51342856+michaelgsharp@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/roslyn-analyzers (#91499)

* Update dependencies from https://github.com/dotnet/roslyn-analyzers build 20230830.1

Microsoft.CodeAnalysis.Analyzers , Microsoft.CodeAnalysis.NetAnalyzers
 From Version 3.11.0-beta1.23428.4 -> To Version 3.11.0-beta1.23430.1

* Update dependencies from https://github.com/dotnet/roslyn-analyzers build 20230902.1

Microsoft.CodeAnalysis.Analyzers , Microsoft.CodeAnalysis.NetAnalyzers
 From Version 3.11.0-beta1.23428.4 -> To Version 3.11.0-beta1.23452.1

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

Rob-Hague committed:
Re-enable an assert in BigInteger (#91158)

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/roslyn (#91240)

* Update dependencies from https://github.com/dotnet/roslyn build 20230828.1

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-2.23428.1

* Update dependencies from https://github.com/dotnet/roslyn build 20230828.2

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-2.23428.2

* Update dependencies from https://github.com/dotnet/roslyn build 20230828.3

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-2.23428.3

* Update dependencies from https://github.com/dotnet/roslyn build 20230828.4

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-2.23428.4

* Update dependencies from https://github.com/dotnet/roslyn build 20230828.6

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23428.6

* Update dependencies from https://github.com/dotnet/roslyn build 20230828.5

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23428.5

* Update dependencies from https://github.com/dotnet/roslyn build 20230828.7

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23428.7

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.1

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23429.1

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.3

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23429.3

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.4

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23429.4

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.6

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23429.6

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.7

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-2.23429.7

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.9

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23429.9

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.10

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23429.10

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.12

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23429.12

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.13

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23429.13

* Update dependencies from https://github.com/dotnet/roslyn build 20230829.14

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23429.14

* Update dependencies from https://github.com/dotnet/roslyn build 20230830.1

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23430.1

* Update dependencies from https://github.com/dotnet/roslyn build 20230830.2

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-2.23430.2

* Update dependencies from https://github.com/dotnet/roslyn build 20230830.3

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23430.3

* Update dependencies from https://github.com/dotnet/roslyn build 20230830.4

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23430.4

* Update dependencies from https://github.com/dotnet/roslyn build 20230830.5

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23430.5

* Update dependencies from https://github.com/dotnet/roslyn build 20230830.6

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23430.6

* Update dependencies from https://github.com/dotnet/roslyn build 20230831.2

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23431.2

* Update dependencies from https://github.com/dotnet/roslyn build 20230831.4

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23431.4

* Update dependencies from https://github.com/dotnet/roslyn build 20230831.6

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23431.6

* Update dependencies from https://github.com/dotnet/roslyn build 20230831.7

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23431.7

* Update dependencies from https://github.com/dotnet/roslyn build 20230901.1

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23451.1

* Update dependencies from https://github.com/dotnet/roslyn build 20230901.2

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23451.2

* Update dependencies from https://github.com/dotnet/roslyn build 20230901.3

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23451.3

* Update dependencies from https://github.com/dotnet/roslyn build 20230901.4

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23451.4

* Update dependencies from https://github.com/dotnet/roslyn build 20230904.1

Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.Net.Compilers.Toolset
 From Version 4.8.0-2.23426.1 -> To Version 4.8.0-3.23454.1

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

matouskozak committed:
remove Vector2 SIMD support from amd64 and wasm if

matouskozak committed:
remove Vector2 SIMD support from amd64 and wasm if

radical committed:
[wasm] Upgrade wasm helix VMs to ubuntu 22.04 (#90881)

* [wasm] Move helix queues to ubuntu 22.04

* Revert https://github.com/dotnet/runtime/pull/90223 + fix v8 tests.

* CI: update other references for wasm helix images

---------

Co-authored-by: Ilona Tomkowicz <itomkowicz@microsoft.com>

steveisok committed:
[maccatalyst] Make sure MacProxy is included in System.Net.Http (#91473)

https://github.com/dotnet/runtime/issues/90258 identified that `HttpClient.DefaultProxy` returned `System.Net.Http.HttpNoProxy` when using MacCatalyst. This is due to a condition in System.Net.Http.csproj not including maccatalyst as a condition and as a result, skipped adding the MacProxy sources.

Way back when, https://github.com/dotnet/runtime/pull/47823 had the condition, but this appears to have changed in https://github.com/dotnet/runtime/pull/48652 when we thought MacCatalyst would inherit settings from iOS. That turned out to not be the case and this check was never added back.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#91546)

Microsoft.Dotnet.Sdk.Internal
 From Version 9.0.100-alpha.1.23451.3 -> To Version 9.0.100-alpha.1.23453.2

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

vargaz committed:
[wasi] Fix pinvoke support. (#91555)

Link NativeReference items into the executable.

layomia committed:
Enable output compilation validation in binder gen unit tests (#91565)

janvorli committed:
Fix handling of `--help` in the build-commons.sh (#91590)

The `--help` was not recognized in that script because double dashes
were converted to single ones before processing the options. That
causes e.g. the src/test/build.sh to not to recognize the `--help`
option, so it instead starts building the tests.

This change fixes it

liveans committed:
Fix Dispose and SendData Race on Http3 Test (#91291)

* Fix Dispose and Send Data Race

* Review feedback

pavelsavara committed:
[browser] ENVIRONMENT_IS_WEB is also true on WebWorker (#91589)

vcsjones committed:
Remove allocation in GetPropertyAsDWord for CNG

jkotas committed:
[NativeAOT] Fix formatting or assertion failures and fail-fasts (#91300)

Fix a long standing TODO. Formatting of assertion failures and fail-fasts in native AOT matches regular CoreCLR with this change.

```csharp
Debug.Assert(false, "Something is not right.");
```

Before:
```
Unhandled Exception: System.Diagnostics.DebugProvider+DebugAssertException: Something is not right.
   at System.Diagnostics.DebugProvider.Fail(String, String) + 0x49
   at System.Diagnostics.Debug.Fail(String, String) + 0x5a
   at System.Diagnostics.Debug.Assert(Boolean, String, String) + 0x2d
   at System.Diagnostics.Debug.Assert(Boolean, String) + 0x28
   at Program.<Main>$(String[] args) + 0x1d
   at test!<BaseAddress>+0x1d3a67
   at test!<BaseAddress>+0x1d3af5
```

After:
```
Process terminated. Assertion failed.
Something is not right.
   at System.Diagnostics.DebugProvider.Fail(String, String) + 0x49
   at System.Diagnostics.Debug.Fail(String, String) + 0x5a
   at System.Diagnostics.Debug.Assert(Boolean, String, String) + 0x2d
   at System.Diagnostics.Debug.Assert(Boolean, String) + 0x28
   at Program.<Main>$(String[] args) + 0x1d
   at repro!<BaseAddress>+0x1d51c7
   at repro!<BaseAddress>+0x1d5255
```

```csharp
Environment.FailFast("Fatal error!");
```

Before:
```
Process terminated. Fatal error!
```

After:
```
Process terminated. Fatal error!
   at System.RuntimeExceptionHelpers.FailFast(String, Exception, String, RhFailFastReason, IntPtr, IntPtr) + 0x335
   at System.Environment.FailFast(String) + 0x35
   at Program.<Main>$(String[] args) + 0x1b
   at repro!<BaseAddress>+0x1d40c7
   at repro!<BaseAddress>+0x1d4155
```

elinor-fung committed:
Include dotnet/hostfxr/hostpolicy symbols in Microsoft.NETCore.App.Host/Runtime packages (#91031)

matouskozak committed:
faddv 64-bit, extract VREG_FULL only

matouskozak committed:
faddv 64-bit, extract VREG_FULL only

tomeksowi committed:
[RISC-V] ELT Profiler Bring-Up (#91313)

* [RISC-V] Generate profiling function callbacks

Initial implementation based on ARM64 code.

* [RISC-V] Fix asm stub for calling profiler callbacks

* Fix argument registers according to RISC-V calling convention
* Fix field offsets for PROFILE_PLATFORM_SPECIFIC_DATA
* Make sure field offsets for PROFILE_PLATFORM_SPECIFIC_DATA stay fixed by static asserting the offsets in asmconstants.h

* [RISC-V] Pass arguments for Profile(Enter|Leave|Tailcall)Naked stubs in t0 and t1 because t2 is used to store the call address of the stub

* [RISC-V] Remove unimplemented definition of EmitRet

* [RISC-V] Copy struct from registers into a buffer so that profiler can see whole struct arguments laid out in memory

* [RISC-V] Implement ProfileArgIterator::GetReturnBufferAddr()

Since the RISC-V ABI says values are returned like the first named argument, re-use the struct copying routine from argument parsing as much as possible.

* Factor out duplicated test results checking routine in SlowPathELTProfiler::Shutdown()

* [RISC-V] Clean up PROFILE_PLATFORM_SPECIFIC_DATA

* Remove unused t0 field
* Remove 'unused' field and widen 'flags' to 64 bits to maintain alignment and shave off one sw instruction

* [RISC-V] Remove commented out code

* [RISC-V] Fix formatting

* [RISC-V] Apply format patch from failed check

* [RISC-V] Post-review fixes

akoeplinger committed:
[mono] Reenable C4244 MSVC warning and fix occurrences (#91395)

Fixes https://github.com/dotnet/runtime/issues/91249

We originally fixed the warnings with https://github.com/dotnet/runtime/pull/69236 but due to `add_compile_options()` calls in zlib.cmake _all_ of the mono native libs - not just the zlib components - were being built with C4244 disabled and a few couple more leaked in.

This fixes the occurrences and reworks the .cmake files a bit to hopefully prevent this in the future.

cshung committed:
Delete the RequiresPreviewFeaturesAttribute from RefreshMemoryLimit (#91622)

elinor-fung committed:
Switch the `AppWithSubDirs` project used for single-file tests to project that is built outside of test runs (#91477)

- Make a installer/tests/Assets/Projects directory that configures projects to build using the live targeting packs
  - Move `AppWithSubDirs` under this directory
- Put test output under `artifacts/tests/host/<os>.<arch>.<config>` instead of `artifacts/tests/<config>` (better aligned with coreclr test output structure)
- Add a `host.pretest` subset for test assets needed for host tests
  - Currently it just builds the one test asset project, but the intention is that more projects can be moved such that they are built outside of test runs. Setting up the shared framework directory could also be part of this.
- Add `SingleFileTestApp` helper that uses output from built test asset projects:
  - Lays out a directory for the non-bundled app
  - Creates bundles based on that directory and the (live) shared framework
    - This should better model the product construction than existing tests, which bundle the entire self-contained output directory (so it has all the native host/runtime files that wouldn't be part of single-file)
  - I had played with merging this with `TestApp` or reusing `TestApp.Populate*`, but this was special enough (files not on disk, only assemblies) that it seemed more confusing to try to share a code path.
- Update `BundledAppWithSubDirs` to use this new mechanism for creating a single-file test app instead of copying/publishing projects
  - On my Windows machine running on Release everything, these go from a bit over a minute to 12 seconds

AndyAyersMS committed:
Fix overly aggressive assert (#91651)

Closes #91349

stephentoub committed:
Fix allocation in DateTime.TryParse when it returns false (#91303)

On some failure paths, it was boxing a char or an int that would have been used as part of creating an exception on a Parse path but that's ignored on a TryParse path. This avoids that boxing by storing those values into a dedicated int field on the DateTimeResult. While doing this, I also consolidated several other fields, such that this not only avoids the boxing, it also results in a net reduction of the size of DateTimeResult (it's passed around by ref, so this only slightly reduces a zeroing burden).

EgorBo committed:
Fix ldapur instruction (#91652)

MichalStrehovsky committed:
Respect StackTraceHiddenAttribute (#91572)

Fixes #91309.

NativeAOT sources stack trace data from two places: reflection metadata (if the method was reflection-visible), or specialized stack trace metadata. Finding out if a method should be hidden from stack traces in the former case is easy: just look for the attribute. The latter case requires compiler work.

In this PR, I'm extending the specialized stack trace metadata format to contain a bit that says if the frame should be hidden or not. I'm doing it in a way that we can also do this for compiler-generated methods (that otherwise don't show up in stack trace metadata).

The runtime behavior is similar to CoreCLR - the methods show up in individual stack frames, but if we stringify the stack trace, they'll be skipped. I'm adding a specialized test that tests the two sources of metadata.

I'm also marking the methods involved in the startup path as stack trace hidden.

There is one thing I skipped - CoreCLR has logic to force generate the stack trace metadata for the first frame. If we do that, the compiler-generated startup code starts showing up in stack traces again. Marking `Main` as stack trace hidden might lead to an empty stack trace.

janvorli committed:
Fix x64 crossbuild on macOS arm64 (#91413)

 cpufeatures.h/.c uses TARGET_ ifdefs, but it should use HOST_ ifdefs

matouskozak committed:
Merge branch 'arm64-vector2-intrinsics' of github.com:matouskozak/runtime into arm64-vector2-intrinsics
